### DISCUSSION TITLE
How to apply transition effects when transitioning from light mode to dark mode and vice verse?

### DISCUSSION THREAD

**Discussion by vinaydawani** [tailwindlabs/tailwindcss#4976] [Help] [answered]
Hey all,

So I'm building a small-ish project with tailwindCSS and thought of implementing the dark mode. I made a button and mapped it to put the `dark` class in the html tag. After testing for a bit, I realized that switching looked kind of odd because it is happening instantaneously. Is there a way to apply a transition duration or timing function to this change?

Here's the basic logic of how I'm handling the change (also I'm using vue):
```JavaScript
<template>
  <!-- <span class="material-icons">&#xE51C;&#xE518;</span> -->
  <input @click="darkClassToggle" id="toggle" class="toggle" type="checkbox" />
</template>

<script>
export default {
  name: "darkModeToggle",
  methods: {
    darkClassToggle() {
      const toggle = document.querySelector(".toggle");
      const html = document.firstElementChild;
      if (toggle.checked) {
        html.classList.remove("dark");
      } else {
        html.classList.add("dark");
      }
    },
  },
};
</script>
```
Thank you for any help

**Comment by dance2die** [comment, marked as answer]
You can add `transition` and `duration` on elements you are changing colors to.

Demo: https://codesandbox.io/s/tailwindcss-dark-mode-transition-ed0kf?file=/public/index.html

**Reply by vinaydawani** [reply]
Thank you so much for this. I got my code running and also figured out what was I doing wrong.
